subprojects/docs/src/docs/userguide/kotlin_dsl.adoc

```diff
@@ -603,165 +483,215 @@ include::sample[dir="kotlinDsl/containers/scope",files="build.gradle.kts[tags=sc
 
-// Reference existing extra properties
-val myProperty: String by extra
-val myNullableProperty: String? by extra
+val myProperty: String by extra  // <3>
+val myNullableProperty: String? by extra  // <4>
 ----
+<1> Creates a new extra property called `myNewProperty` in the current context (the project in this case) and initializes it with the value `"initial value"`, which also determines the property's _type_
+<2> Create a new extra property whose initial value is calculated when the property is accessed
+<3> Binds an existing extra property from the current context (the project in this case) to a `myProperty` reference
+<4> Does the same as the previous line but allows the property to have a null value
 
-Let's say the snippet above is from your root project build script, in order to reference the `myNewProperty` from a sub project you'd do:
+This approach works for all Gradle scripts: project build scripts, script plugins, settings scripts and initialization scripts.
+
+You can also access extra properties on a root project from a subproject using the following syntax:
 
 .my-sub-project/build.gradle.kts
 [source,kotlin]
 ----
-val myNewProperty: String by rootProject.extra
+val myNewProperty: String by rootProject.extra  // <1>
 ----
+<1> Binds the root project's `myNewProperty` extra property to a reference of the same name
 
-It works the same way in settings or init scripts.
-
-All `ExtensionAware` types can host extra properties, `Task` is a common example:
+Extra properties aren't just limited to projects.
+For example, `Task` extends `ExtensionAware`, so you can attach extra properties to tasks as well.
+Here's an example that defines a new `myNewTaskProperty` on the `test` task and then uses that property to initialize another task:
 
 .build.gradle.kts
 [source,kotlin]
 ----
 tasks {
     test {
-        // Define new extra properties
-        val myNewTaskProperty by extra("initial value")
-        val myOtherNewTaskProperty by extra { "lazy initial value" }
+        val reportType by extra("dev")  // <1>
+        doLast {
+            // Use 'suffix' for post processing of reports
+        }
+    }
 
-        // Reference existing extra properties
-        val myTaskProperty: String by extra
-        val myNullableTaskProperty: String? by extra
+    register<Zip>("archiveTestReports") {
+        val reportType: String by test.get().extra  // <2>
+        appendix = reportType
+        from(test.get().reports.html.destination)
     }
 }
 ----
+<1> Creates a new `reportType` extra property on the `test` task
+<2> Makes the `test` task's `reportType` extra property available to configure the `archiveTestReports` task
+
+If you're happy to use eager configuration rather than the configuration avoidance APIs, you could use a single, "global" property for the report type, like this:
+
+.build.gradle.kts
+[source,kotlin]
+----
+tasks.test.doLast { ... }
+
+val testReportType by tasks.test.extra("dev")  // <1>
```

Comment:
:x: should be `val testReportType by tasks.test.get().extra("dev")`

Reply:
Ah, this was the one I didn't test. Ideally, all the samples in this chapter should have tests, but I don't feel up to it right now.